src/visualizations/config/adapters/dhis_highcharts/getFilterTitle.js

```diff
@@ -1,44 +1,99 @@
 import isArray from 'd2-utilizr/lib/isArray'
+import i18n from '@dhis2/d2-i18n'
+import { ouIdHelper } from '../../../../modules/ouIdHelper'
+import { dimensionIs } from '../../../../modules/layout/dimensionIs'
+import { DIMENSION_ID_ORGUNIT } from '../../../../modules/fixedDimensions'
+import { dimensionGetItems } from '../../../../modules/layout/dimensionGetItems'
 
 export default function(filters, metaData) {
-    let title
+    if (isArray(filters)) {
+        const titleParts = []
 
-    if (isArray(filters) && filters.length) {
-        title = ''
+        filters.forEach(filter => {
+            const items = dimensionGetItems(filter)
+            const hasOuLevel =
+                dimensionIs(filter, DIMENSION_ID_ORGUNIT) &&
+                items.some(item => ouIdHelper.hasLevelPrefix(item.id))
+            const hasOuGroup =
+                dimensionIs(filter, DIMENSION_ID_ORGUNIT) &&
+                items.some(item => ouIdHelper.hasGroupPrefix(item.id))
 
-        let filterItems, i, l
+            if (hasOuLevel || hasOuGroup) {
+                if (hasOuGroup) {
+                    titleParts.push(getOuTitle(items, metaData, false))
+                }
 
-        filters.forEach((dimension, index, array) => {
-            filterItems = metaData.dimensions[dimension.dimension]
+                if (hasOuLevel) {
+                    titleParts.push(getOuTitle(items, metaData, true))
+                }
+            } else {
+                const filterItems = metaData.dimensions[filter.dimension]
 
-            if (isArray(filterItems)) {
-                l = filterItems.length
-                let id
+                if (isArray(filterItems)) {
+                    const hasMetadata = filterItems.some(id =>
```

Comment:
I'm not sure whether this is a safe change.  Before, it checked the metadata for each item. Now it just checks that at least one item is present in the metadata.

Reply:
Restoring the "old" way